spec/Section 4 -- Composition.md

```diff
@@ -341,12 +341,12 @@ ERROR
       - For each {argumentName} in {argumentNames}
         - Let {arguments} be the set of all arguments with the name
           {argumentName} from all fields in {fields}.
-        - Let {defaultValue} be the first default value found in {arguments}.
+        - Let {defaultValues} be all the default values found in {arguments}.
         - Let {externalArguments} be the set of all arguments with the name
           {argumentName} from all fields in {externalFields}.
         - For each {externalArgument} in {externalArguments}
-          - The default value of {externalArgument} must be equal to
-            {defaultValue}.
+          - The default value of {externalArgument} must be equal to all
+            {defaultValues}.
```

Comment:
Assume defaults:

Schema A: 1
Schema B: 2
Schema C: 3 (external)

Wouldn't this compare:

3 with 1
3 with 2
3 with 3 (redundant)

i.e. External defaults shouldn't be included in `defaultValues`?

Reply:
does this change the outcome of the rule?

I more or less try to word the rules as simple as possible.

The actualy implementation of the rule would be completely different anyway i believe.

Reply:
Hmm. I'm not sure. Maybe not in this specific case.

If there were more than 1 external argument, it would fail if the default values of 2 or more external arguments didn't match?

Assume defaults:

Schema A: 2
Schema B: 2 (external)
Schema C: 3 (external)

Compares:

2 with 2 (B - A)
2 with 2 (B - B)
2 with 3 (B - C [external - external])

3 with 2 (C - A)
3 with 2 (C - B [external - external])
3 with 3 (C - C)

Also the explanatory text is:

> This rule ensures that arguments on fields marked as `@external` have default
values compatible with the corresponding arguments on fields from **other source
schemas where the field is defined (non-`@external`)**

🤷‍♂️